Azure-Samples/assistant-data-openai-python-promptflow
# Feature / Added Support for Dev Containers (GitHub Codespaces)

## Purpose


Dev Containers provides configuration-as-code support for developer environment setup.
Launch repo in GitHub Codespaces (cloud) or Docker Desktop (local) for quickstart setup. This PR adds the `.devcontainer/` folder with initial configuration files that match other azd templates for AI projects. 

A second (minor) change involves making postprovision scripts executable so they can be automatically run as azd hooks from the command-line in a dev container environment

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Click the "Code" tab in GitHub repository (in browser) to launch the repository.
If container setup is valid, you will get a VS Code editor (in browser) with Docker container runtime backend (in cloud) with dev environment already setup.

Once the Codespaces is launched, you can complete the setup in 2 steps:

```
azd auth login
azd up
```


* Test the code


The test of the code lies in the ability to see a working Visual Studio Code editor environment connected to a Docker instance running in cloud (GitHub Codespaces) or local (Docker Desktop).

## What to Check
Verify that your environment has the necessary tools installed correctly.
* Python 3.11+ - `python --version`
* Azure Developer CLI 1.9+ - `azd version`
* Azure CLI 2.6+ - `az version`
* Prompt Flow 1.11+ - `pf version`
